editor/editor_help.cpp

```diff
@@ -225,35 +272,54 @@ void EditorHelp::_add_type(const String &p_type, const String &p_enum) {
 	bool is_enum_type = !p_enum.is_empty();
 	bool can_ref = !p_type.contains("*") || is_enum_type;
 
-	String t = p_type;
+	String link_t = p_type; // For links in metadata
+	String display_t = link_t; // For display purposes
 	if (is_enum_type) {
-		if (p_enum.get_slice_count(".") > 1) {
-			t = p_enum.get_slice(".", 1);
-		} else {
-			t = p_enum.get_slice(".", 0);
+		link_t = p_enum; // The link for enums is always the full enum description
+		Vector<String> slices = p_enum.split(".");
+
+		if (slices.size() == 0) { // This is a local enum with no class specifier
+			display_t = link_t;
+		} else { // There is a class specifier
+			// Replace :: from enum's use of fully qualified class names with regular .
+			slices.set(0, slices[0].replace("::", "."));
+			link_t = String(".").join(slices);
+
+			// Remove prefix up to and including current class name. This may make class specifier empty.
+			slices.set(0, _contextualize_class_specifier(slices[0], edited_class, false));
+			if (slices[0].is_empty()) {
+				slices.remove_at(0);
+			}
+			display_t = String(".").join(slices); // Use "." to separate all classes & enums
+		}
+	} else {
+		// Remove current class specifier from original type description
+		display_t = _contextualize_class_specifier(p_type, edited_class, true);
+		if (display_t.is_empty()) {
```

Comment:
I don't think it's possible for this to be empty at this point, with keep last being set to true, is it? I'm not against an extra check regardless, but perhaps would add a comment that this is unlikely and is done just in case.

Reply:
You are right! :) It was a last minute change to the contextualize function and I forgot to remove those lines! Nice catch!